### PR TITLE
Implement Dijkstra's algorithm

### PR DESCRIPTION
**Pull request by songofroland**

### CODE REVIEW

File: src/algo/dijkstra.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    let unvisitedNodes: Nodes = {};
+    for (const [key, value] of Object.entries(nodes)) {
+        if (unvisited.has(parseInt(key))) {
+            unvisitedNodes[key] = value;
+        }
+    }
+    return unvisitedNodes;
+}
+
+function getCheapesNode(nodes: Nodes) {
```
Comment (wilmhit) [inline src/algo/dijkstra.ts:61]: Cheapest?

File: src/algo/dijkstra.test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+]
+
+const expectedDistances = {
+    '0': 0,
+    '1': 2,
+    '2': 6,
+    '3': 7,
+    '4': 17,
+    '5': 22,
+    '6': 19
```
Comment (wilmhit) [inline src/algo/dijkstra.test.ts:22]: ,

File: src/algo/dijkstra.test.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+const expectedPaths = {
+    '0': [],
+    '1': [0],
+    '2': [0],
+    '3': [0, 1],
+    '4': [0, 1, 3],
+    '5': [0, 1, 3],
+    '6': [0, 1, 3, 4]
+}
```
Comment (wilmhit) [inline src/algo/dijkstra.test.ts:33]: ;

File: src/algo/dijkstra.test.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+// TODO: Move to "tests/" directory, test more graphs
+const myGraph: Array<Array<number>> = [
+    [0, 2, 6, 0, 0, 0, 0],
+    [2, 0, 0, 5, 0, 0, 0],
+    [6, 0, 0, 8, 0, 0, 0],
+    [0, 5, 8, 0, 10, 15, 0],
+    [0, 0, 0, 10, 0, 6, 2],
+    [0, 0, 0, 15, 6, 0, 6],
+    [0, 0, 0, 0, 2, 6, 0],
+]
```
Comment (wilmhit) [inline src/algo/dijkstra.test.ts:13]: ;

File: src/algo/dijkstra.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    let unvisitedNodes: Set<number> = new Set(nodes);
+
+    costs[startNode] = 0;
+    let currentNode = startNode;
+
+    while (unvisitedNodes.size) {
+        for (const [neighboredNode, neighbourCost] of graph[currentNode].entries()) {
+            if (neighbourCost !== 0) {
+                let previousCost = costs[neighboredNode];
+                let currentCost = neighbourCost + costs[currentNode];
```
Comment (wilmhit) [inline src/algo/dijkstra.ts:35]: previousCost, currentCost can be const?

**Review by wilmhit — CHANGES_REQUESTED** [review]
Missing semicolon or "let" instead of "const" should cause compilation errors with tsc. Are you sure you have correct configuration locally?
Also check for missing lines at EOF.

**Comment by songofroland** [comment]
> Missing semicolon or "let" instead of "const" should cause compilation errors with tsc. Are you sure you have correct configuration locally?

Good catch, my eslint was not present at all.

**Review by wilmhit — APPROVED** [review]

**Pull request merged**